.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,36 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '29 7 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
```

Comment:
I think there should be a typescript version of it https://codeql.github.com/docs/codeql-overview/supported-languages-and-frameworks/#languages-and-compilers

Reply:
interesting. the docs are not consistent here 
https://codeql.github.com/docs/codeql-language-guides/codeql-library-for-typescript/

> [CodeQL libraries for JavaScript](https://codeql.github.com/docs/codeql-language-guides/codeql-library-for-javascript/) covers most of this library, and is also relevant for TypeScript analysis. This document supplements the JavaScript documentation with the TypeScript-specific classes and predicates.

Reply:
the workflow is running a build, resulting in js. I think we're probably fine?

Reply:
If it works, it should be fine :) I am a bit insecure because it checks [here](https://github.com/github/codeql-action/blob/main/lib/languages.js#L39) for `["cpp", "java", "csharp", "swift"]`

Reply:
I can't come up with a consistent answer tbh. Official recommendations are js, but people seem to be doing [js, ts] as well. if this does not fail, I see no harm in adding ts
https://github.com/search?q=CodeQL+typescript+matrix&type=Code

Reply:
reverting. this answer looks official and is in line with the docs https://github.com/github/codeql-action/issues/365